src/services/company.js

```diff
@@ -99,40 +105,86 @@ class CompanyService {
      * @param {*} attributes object containing the attributes to change in company
      */
     changeAttributes(company_id, attributes) {
-        return Company.findOneAndUpdate({ _id: company_id }, attributes);
+        return Company.findOneAndUpdate(
+            { _id: company_id },
+            attributes,
+            { new: true, omitUndefined: true },
+            (err) => {
+                if (err) {
+                    console.error(err);
+                    throw err;
+                }
+            });
     }
 
-    async sendCompanyBlockedNotification(companyId) {
+    /**
+     * E-mails the given company using the provided notification template.
+     * @param {*} companyId the id of the company to whom the notifications is sent
+     * @param {*} notification the notification to send
+     */
+    async _sendCompanyNotification(companyId, notification) {
         try {
             const company = await Company.findById(companyId);
             const companyAccount = await Account.findOne({
                 company
             });
             await EmailService.sendMail({
                 to: companyAccount.email,
-                ...COMPANY_BLOCKED_NOTIFICATION(company.name),
+                ...notification(company.name),
             });
         } catch (err) {
             console.error(err);
             throw err;
         }
     }
 
+    async sendCompanyBlockedNotification(companyId) {
+        await this._sendCompanyNotification(companyId, COMPANY_BLOCKED_NOTIFICATION);
+    }
+
     async sendCompanyUnblockedNotification(companyId) {
+        await this._sendCompanyNotification(companyId, COMPANY_UNBLOCKED_NOTIFICATION);
+    }
+
+    disable(companyId) {
+
         try {
-            const company = await Company.findById(companyId);
-            const companyAccount = await Account.findOne({
-                company
-            });
-            await EmailService.sendMail({
-                to: companyAccount.email,
-                ...COMPANY_UNBLOCKED_NOTIFICATION(company.name),
-            });
+            return this.changeAttributes(
+                companyId,
+                {
+                    isDisabled: true,
+                }
+            );
```

Comment:
if this fits in a line, it's better to avoid taking 6 for an easier read
```suggestion
            return this.changeAttributes(companyId, { isDisabled: true });
```